pymc_marketing/mmm/models/components/lagging.py

```diff
@@ -0,0 +1,394 @@
+from typing import Dict, Optional, Union
+
+import pymc as pm
+
+from pymc_marketing.mmm.transformers import geometric_adstock, weibull_adstock
+from pymc_marketing.mmm.utils import _get_distribution
+
+
+class BaseFunction:
+    def __init__(self, max_lagging: int = 4, model: Optional[pm.Model] = None):
```

Comment:
Does `model` have to be passed into init? Or can it just be passed into `apply` instead. 
I think carrying around to just use in one method can become cumbersome

Reply:
This might be repeat comment, but if we assume that the apply happens in a model context, then we don't need to pass the model at all and the class doesn't need it anymore